scripts/tests/test_zkai_native_seq32_attention_mlp_adjacent_probe_b_bucket_attribution_gate.py

```
@@ -0,0 +1,183 @@
+            tmp_path = pathlib.Path(tmp)
+            with self.assertRaisesRegex(
+                self.gate.AdjacentProbeBBucketAttributionGateError,
+                "output path escapes evidence dir",
+            ):
+                self.gate.write_outputs(tmp_path / "out.json", tmp_path / "out.tsv", self.payload)
+
+    def test_rejects_non_regular_repo_file_before_reading(self):
+        with tempfile.TemporaryDirectory(dir=self.gate.EVIDENCE_DIR) as tmp:
+            directory_path = pathlib.Path(tmp) / "not-a-file"
+            directory_path.mkdir()
+            with self.assertRaisesRegex(
```

Ruff (0.15.13)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)